File: ruby/j10.arguments.rb
#!/usr/bin/ruby
arg_count = ARGV.length   # get num of arguments
first     = 0             # set index of first element
last      = arg_count - 1 # set index of last element

puts "The arguments passed are (reverse order):"
# iterative loop using array index
for index in first.upto(last)             # decrement from # of args to 0 
  puts " item #{index+1}: #{ARGV[index]}" # count & correponding ARGV element
end

File: php/j10.arguments.php
#!/usr/bin/php
<?php
$arg_count = $argc - 1;  // get num of arguments
$first     = 1;          // set index of initial element
$last      = $arg_count; // set index of last element

echo "The arguments passed are:\n";
// iterative loop to enumerate args
for ( $count = 1; $count <= $last ; $count++ )
  // output count and argument using count index
  echo "  item $count: $argv[$count]\n";
?>